src/flacarray/array.py

```diff
@@ -289,7 +289,15 @@ def __getitem__(self, key):
             first_stream_sample=first,
             last_stream_sample=last,
         )
-        return arr
+        arr_indices = np.transpose(np.array(strm_indices))
+        leading = [
+            len(np.unique(arr_indices[x])) for x in range(len(self._leading_shape))
+        ]
+        if first is None:
+            full_shape = tuple(leading) + (self._shape[-1],)
+        else:
+            full_shape = tuple(leading) + (last - first,)
+        return np.squeeze(arr.reshape(full_shape))
```

Comment:
I don't understand this fully but the use of squeeze is suspicious... what if the thing I'm asking for has a legitimate dimension of length 1?  E.g. `something[0,:1,:]`?

To convince me ... add a test case that includes `slice(0, 1)`.

Reply:
This was a really good point, and indeed the output dimensions did not match expectations.  I have now overhauled `__getitem__()` and expanded the unit tests to compare the dimensions of the sliced output to the corresponding dimensions from slicing the input numpy array.  Thanks for re-reviewing at your convenience.